presto-native-sidecar-plugin/src/main/java/com/facebook/presto/sidecar/typemanager/NativeTypeManager.java

```diff
@@ -0,0 +1,195 @@
+import static com.facebook.presto.common.type.StandardTypes.HYPER_LOG_LOG;
+import static com.facebook.presto.common.type.StandardTypes.INTEGER;
+import static com.facebook.presto.common.type.StandardTypes.INTERVAL_DAY_TO_SECOND;
+import static com.facebook.presto.common.type.StandardTypes.INTERVAL_YEAR_TO_MONTH;
+import static com.facebook.presto.common.type.StandardTypes.IPADDRESS;
+import static com.facebook.presto.common.type.StandardTypes.IPPREFIX;
+import static com.facebook.presto.common.type.StandardTypes.JSON;
+import static com.facebook.presto.common.type.StandardTypes.MAP;
+import static com.facebook.presto.common.type.StandardTypes.REAL;
+import static com.facebook.presto.common.type.StandardTypes.ROW;
+import static com.facebook.presto.common.type.StandardTypes.SMALLINT;
+import static com.facebook.presto.common.type.StandardTypes.TIMESTAMP;
+import static com.facebook.presto.common.type.StandardTypes.TIMESTAMP_WITH_TIME_ZONE;
+import static com.facebook.presto.common.type.StandardTypes.TINYINT;
+import static com.facebook.presto.common.type.StandardTypes.UUID;
+import static com.facebook.presto.common.type.StandardTypes.VARBINARY;
+import static com.facebook.presto.common.type.StandardTypes.VARCHAR;
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.HOURS;
+
+public class NativeTypeManager
+        implements TypesProvider
+{
+    private static final Logger log = Logger.get(NativeTypeManager.class);
+
+    private static final Set<String> NATIVE_ENGINE_SUPPORTED_TYPES =
+            ImmutableSet.of(
+                    BIGINT,
+                    REAL,
+                    VARBINARY,
+                    TIMESTAMP,
+                    TINYINT,
+                    BOOLEAN,
+                    DATE,
+                    INTEGER,
+                    DOUBLE,
+                    SMALLINT,
+                    HYPER_LOG_LOG,
+                    JSON,
+                    TIMESTAMP_WITH_TIME_ZONE,
+                    UUID,
+                    IPADDRESS,
+                    IPPREFIX,
+                    INTERVAL_DAY_TO_SECOND,
+                    INTERVAL_YEAR_TO_MONTH,
+                    VARCHAR);
+
+    private static final Set<String> NATIVE_ENGINE_SUPPORTED_PARAMETRIC_TYPES =
+            ImmutableSet.of(
+                    ARRAY,
+                    DECIMAL,
+                    MAP,
+                    ROW,
+                    FUNCTION);
+
+    private final TypeManager typeManager;
+    private final LoadingCache<ExactTypeSignature, Type> parametricTypeCache;
```

Comment:
Just copy the `ExactTypeSignature` here as well.  It's a bit too specific to be useful as a general API in common.

Reply:
The problem is `ExactTypeSignature` is used in `FunctionAndTypeManager` too.

Reply:
Done.